typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -113,8 +121,103 @@ export class EvmHookModule extends HyperlaneModule<
-    throw new Error('Method not implemented.');
+  public async update(
+    targetConfig: HookConfig,
+  ): Promise<AnnotatedEV5Transaction[]> {
+    // save current config for comparison
+    // normalize the config to ensure it's in a consistent format for comparison
+    const currentConfig = normalizeConfig(await this.read());
+
+    // Update the config
+    this.args.config = targetConfig;
+
+    // If configs match, no updates needed
+    if (configDeepEquals(currentConfig, targetConfig)) {
+      return [];
+    }
+
+    // Else, we have to figure out what an update for this Hook entails
+
+    // If target config is an address Hook, just update the address
+    // if config -> address Hook, update address
+    // if address Hook -> address Hook, update address
+    if (typeof targetConfig === 'string') {
+      // TODO: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3773
+      this.args.addresses.deployedHook = targetConfig;
+      return [];
+    }
+
+    // Check if we need to deploy a new Hook
+    if (
+      // if address Hook -> config, do a new deploy
+      typeof currentConfig === 'string' ||
+      // if config -> config, AND types are different, do a new deploy
+      currentConfig.type !== targetConfig.type ||
+      // if it is an immutable Hook, do a new deploy
+      IMMUTABLE_HOOK_TYPE.includes(targetConfig.type)
+    ) {
+      const contract = await this.deploy({
+        config: targetConfig,
+      });
+
+      this.args.addresses.deployedHook = contract.address;
+      return [];
+    }
+
+    let updateTxs: AnnotatedEV5Transaction[];
+
+    // obtain the update txs for each hook type
+    switch (targetConfig.type) {
+      case HookType.INTERCHAIN_GAS_PAYMASTER:
+        updateTxs = await this.updateIgpHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      case HookType.PROTOCOL_FEE:
+        updateTxs = await this.updateProtocolFeeHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      case HookType.PAUSABLE:
+        updateTxs = await this.updatePausableHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      case HookType.ROUTING:
+      case HookType.FALLBACK_ROUTING:
+        updateTxs = await this.updateRoutingHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      default:
+        // MERKLE_TREE, AGGREGATION and OP_STACK hooks should already be handled before the switch
+        throw new Error(`Unsupported hook type: ${targetConfig.type}`);
+    }
+
+    // Lastly, check if the resolved owner is different from the current owner
+    const owner = await Ownable__factory.connect(
+      this.args.addresses.deployedHook,
+      this.multiProvider.getProvider(this.chain),
+    ).owner();
+
+    // Return an ownership transfer transaction if required
+    if (!eqAddress(targetConfig.owner, owner)) {
```

Comment:
In which case will this execute? i thought the owner is always transferred, if needed in the deploy fns?

Reply:
this can be super useful down the line with checker/governance actions too